astropy/coordinates/tests/test_sites.py

```diff
@@ -113,3 +113,14 @@ def test_registry():
 
     loc2 = reg['sIte a']
     assert loc2 is loc
+
+@remote_data
+def test_non_EarthLocation():
+    """
+    A regression test for a typo bug pointed out at the bottom of
+    https://github.com/astropy/astropy/pull/4042
+    """
+    class EarthLocation2(EarthLocation):
+        pass
+
+    EarthLocation2.of_site('keck')
```

Comment:
Perhaps overkill, but maybe good to test it not only works, but does what it is expected to:
```
el2 = EarthLocation2.of_site('keck')
assert type(el2) is EarthLocation2
assert el2.info.name == 'keck'
```
With that, feel free to merge...